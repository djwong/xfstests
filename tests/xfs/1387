#! /bin/bash
# FS QA Test No. 1387
#
# Populate a XFS filesystem and fuzz every superblock field.
# Do not fix the filesystem, to test metadata verifiers.
#
#-----------------------------------------------------------------------
# Copyright (c) 2018 Oracle, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 7 15

_cleanup()
{
	cd /
	rm -rf $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/populate
. ./common/fuzzy

# real QA test starts here
_supported_os Linux
_supported_fs xfs
_require_scratch_xfs_fuzz_fields

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

echo "Fuzz superblock"
_scratch_xfs_fuzz_metadata '' 'none' 'sb 1' >> $seqres.full
echo "Done fuzzing superblock"

# success, all done
status=0
exit
