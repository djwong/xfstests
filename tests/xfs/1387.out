QA output created by 1387
Format and populate
Fuzz superblock
re-repair failed (1) with magicnum = zeroes.
re-repair failed (1) with magicnum = ones.
re-repair failed (1) with magicnum = firstbit.
re-repair failed (1) with magicnum = middlebit.
re-repair failed (1) with magicnum = lastbit.
re-repair failed (1) with magicnum = add.
re-repair failed (1) with magicnum = sub.
re-repair failed (1) with magicnum = random.
re-repair failed (1) with blocksize = zeroes.
re-repair failed (1) with blocksize = ones.
re-repair failed (1) with blocksize = firstbit.
re-repair failed (1) with blocksize = middlebit.
re-repair failed (1) with blocksize = lastbit.
re-repair failed (1) with blocksize = add.
re-repair failed (1) with blocksize = sub.
re-repair failed (1) with blocksize = random.
re-repair failed (1) with dblocks = zeroes.
re-repair failed (1) with dblocks = ones.
re-repair failed (1) with dblocks = firstbit.
re-repair failed (1) with dblocks = middlebit.
re-repair failed (1) with dblocks = lastbit.
re-repair failed (1) with dblocks = add.
re-repair failed (1) with dblocks = sub.
re-repair failed (1) with dblocks = random.
re-repair failed (1) with rblocks = ones.
re-repair failed (1) with rblocks = firstbit.
re-repair failed (1) with rblocks = middlebit.
re-repair failed (1) with rblocks = lastbit.
re-repair failed (1) with rblocks = add.
re-repair failed (1) with rblocks = sub.
re-repair failed (1) with rblocks = random.
re-repair failed (1) with rextents = ones.
re-repair failed (1) with rextents = firstbit.
re-repair failed (1) with rextents = middlebit.
re-repair failed (1) with rextents = lastbit.
re-repair failed (1) with rextents = add.
re-repair failed (1) with rextents = sub.
re-repair failed (1) with rextents = random.
re-repair failed (1) with logstart = zeroes.
re-repair failed (1) with logstart = ones.
re-repair failed (1) with logstart = firstbit.
re-repair failed (1) with logstart = middlebit.
re-repair failed (1) with logstart = lastbit.
re-repair failed (1) with logstart = add.
re-repair failed (1) with logstart = sub.
re-repair failed (1) with logstart = random.
re-repair failed (1) with rextsize = zeroes.
re-repair failed (1) with rextsize = ones.
re-repair failed (1) with rextsize = firstbit.
re-repair failed (1) with rextsize = middlebit.
re-repair failed (1) with rextsize = lastbit.
re-repair failed (1) with rextsize = add.
re-repair failed (1) with rextsize = sub.
re-repair failed (1) with rextsize = random.
re-repair failed (1) with agblocks = zeroes.
re-repair failed (1) with agblocks = ones.
re-repair failed (1) with agblocks = firstbit.
re-repair failed (1) with agblocks = middlebit.
re-repair failed (1) with agblocks = lastbit.
re-repair failed (1) with agblocks = add.
re-repair failed (1) with agblocks = sub.
re-repair failed (1) with agblocks = random.
re-repair failed (1) with agcount = zeroes.
re-repair failed (1) with agcount = ones.
re-repair failed (1) with agcount = firstbit.
re-repair failed (1) with agcount = middlebit.
re-repair failed (1) with agcount = lastbit.
re-repair failed (1) with agcount = add.
re-repair failed (1) with agcount = sub.
re-repair failed (1) with agcount = random.
re-repair failed (1) with rbmblocks = ones.
re-repair failed (1) with rbmblocks = firstbit.
re-repair failed (1) with rbmblocks = middlebit.
re-repair failed (1) with rbmblocks = lastbit.
re-repair failed (1) with rbmblocks = add.
re-repair failed (1) with rbmblocks = sub.
re-repair failed (1) with rbmblocks = random.
re-repair failed (1) with logblocks = zeroes.
re-repair failed (1) with logblocks = ones.
re-repair failed (1) with logblocks = firstbit.
re-repair failed (1) with logblocks = middlebit.
re-repair failed (1) with logblocks = lastbit.
re-repair failed (1) with logblocks = add.
re-repair failed (1) with logblocks = sub.
re-repair failed (1) with logblocks = random.
re-repair failed (1) with versionnum = zeroes.
re-repair failed (1) with versionnum = ones.
re-repair failed (1) with versionnum = lastbit.
re-repair failed (1) with versionnum = add.
re-repair failed (1) with versionnum = sub.
re-repair failed (1) with versionnum = random.
re-repair failed (1) with sectsize = zeroes.
re-repair failed (1) with sectsize = ones.
re-repair failed (1) with sectsize = firstbit.
re-repair failed (1) with sectsize = middlebit.
re-repair failed (1) with sectsize = lastbit.
re-repair failed (1) with sectsize = add.
re-repair failed (1) with sectsize = sub.
re-repair failed (1) with sectsize = random.
re-repair failed (1) with inodesize = zeroes.
re-repair failed (1) with inodesize = ones.
re-repair failed (1) with inodesize = firstbit.
re-repair failed (1) with inodesize = middlebit.
re-repair failed (1) with inodesize = lastbit.
re-repair failed (1) with inodesize = add.
re-repair failed (1) with inodesize = sub.
re-repair failed (1) with inodesize = random.
re-repair failed (1) with inopblock = zeroes.
re-repair failed (1) with inopblock = ones.
re-repair failed (1) with inopblock = firstbit.
re-repair failed (1) with inopblock = middlebit.
re-repair failed (1) with inopblock = lastbit.
re-repair failed (1) with inopblock = add.
re-repair failed (1) with inopblock = sub.
re-repair failed (1) with inopblock = random.
re-repair failed (1) with blocklog = zeroes.
re-repair failed (1) with blocklog = ones.
re-repair failed (1) with blocklog = firstbit.
re-repair failed (1) with blocklog = middlebit.
re-repair failed (1) with blocklog = lastbit.
re-repair failed (1) with blocklog = add.
re-repair failed (1) with blocklog = sub.
re-repair failed (1) with blocklog = random.
re-repair failed (1) with sectlog = zeroes.
re-repair failed (1) with sectlog = ones.
re-repair failed (1) with sectlog = firstbit.
re-repair failed (1) with sectlog = middlebit.
re-repair failed (1) with sectlog = lastbit.
re-repair failed (1) with sectlog = add.
re-repair failed (1) with sectlog = sub.
re-repair failed (1) with sectlog = random.
re-repair failed (1) with inodelog = zeroes.
re-repair failed (1) with inodelog = ones.
re-repair failed (1) with inodelog = firstbit.
re-repair failed (1) with inodelog = middlebit.
re-repair failed (1) with inodelog = lastbit.
re-repair failed (1) with inodelog = add.
re-repair failed (1) with inodelog = sub.
re-repair failed (1) with inodelog = random.
re-repair failed (1) with inopblog = zeroes.
re-repair failed (1) with inopblog = ones.
re-repair failed (1) with inopblog = firstbit.
re-repair failed (1) with inopblog = middlebit.
re-repair failed (1) with inopblog = lastbit.
re-repair failed (1) with inopblog = add.
re-repair failed (1) with inopblog = sub.
re-repair failed (1) with inopblog = random.
re-repair failed (1) with agblklog = zeroes.
re-repair failed (1) with agblklog = ones.
re-repair failed (1) with agblklog = firstbit.
re-repair failed (1) with agblklog = middlebit.
re-repair failed (1) with agblklog = lastbit.
re-repair failed (1) with agblklog = add.
re-repair failed (1) with agblklog = sub.
re-repair failed (1) with agblklog = random.
re-repair failed (1) with rextslog = ones.
re-repair failed (1) with rextslog = firstbit.
re-repair failed (1) with rextslog = middlebit.
re-repair failed (1) with rextslog = lastbit.
re-repair failed (1) with rextslog = add.
re-repair failed (1) with rextslog = sub.
re-repair failed (1) with rextslog = random.
re-repair failed (1) with imax_pct = ones.
re-repair failed (1) with imax_pct = firstbit.
re-repair failed (1) with imax_pct = add.
re-repair failed (1) with imax_pct = sub.
re-repair failed (1) with imax_pct = random.
re-repair failed (1) with frextents = ones.
re-repair failed (1) with frextents = firstbit.
re-repair failed (1) with frextents = middlebit.
re-repair failed (1) with frextents = lastbit.
re-repair failed (1) with frextents = add.
re-repair failed (1) with frextents = sub.
re-repair failed (1) with frextents = random.
re-repair failed (1) with uquotino = firstbit.
re-repair failed (1) with uquotino = middlebit.
re-repair failed (1) with uquotino = lastbit.
re-repair failed (1) with uquotino = add.
re-repair failed (1) with uquotino = sub.
re-repair failed (1) with uquotino = random.
re-repair failed (1) with gquotino = firstbit.
re-repair failed (1) with gquotino = middlebit.
re-repair failed (1) with gquotino = lastbit.
re-repair failed (1) with gquotino = add.
re-repair failed (1) with gquotino = sub.
re-repair failed (1) with gquotino = random.
re-repair failed (1) with qflags = ones.
re-repair failed (1) with qflags = firstbit.
re-repair failed (1) with qflags = middlebit.
re-repair failed (1) with qflags = lastbit.
re-repair failed (1) with qflags = add.
re-repair failed (1) with qflags = sub.
re-repair failed (1) with qflags = random.
re-repair failed (1) with flags = ones.
re-repair failed (1) with flags = firstbit.
re-repair failed (1) with flags = middlebit.
re-repair failed (1) with flags = lastbit.
re-repair failed (1) with flags = add.
re-repair failed (1) with flags = sub.
re-repair failed (1) with flags = random.
re-repair failed (1) with inoalignmt = zeroes.
re-repair failed (1) with inoalignmt = ones.
re-repair failed (1) with inoalignmt = firstbit.
re-repair failed (1) with inoalignmt = middlebit.
re-repair failed (1) with inoalignmt = lastbit.
re-repair failed (1) with inoalignmt = add.
re-repair failed (1) with inoalignmt = sub.
re-repair failed (1) with inoalignmt = random.
re-repair failed (1) with unit = ones.
re-repair failed (1) with unit = firstbit.
re-repair failed (1) with unit = middlebit.
re-repair failed (1) with unit = lastbit.
re-repair failed (1) with unit = add.
re-repair failed (1) with unit = sub.
re-repair failed (1) with unit = random.
re-repair failed (1) with width = ones.
re-repair failed (1) with width = firstbit.
re-repair failed (1) with width = middlebit.
re-repair failed (1) with width = lastbit.
re-repair failed (1) with width = add.
re-repair failed (1) with width = sub.
re-repair failed (1) with width = random.
re-repair failed (1) with dirblklog = ones.
re-repair failed (1) with dirblklog = firstbit.
re-repair failed (1) with dirblklog = middlebit.
re-repair failed (1) with dirblklog = add.
re-repair failed (1) with dirblklog = sub.
re-repair failed (1) with dirblklog = random.
re-repair failed (1) with logsectlog = zeroes.
re-repair failed (1) with logsectlog = ones.
re-repair failed (1) with logsectlog = firstbit.
re-repair failed (1) with logsectlog = middlebit.
re-repair failed (1) with logsectlog = lastbit.
re-repair failed (1) with logsectlog = add.
re-repair failed (1) with logsectlog = sub.
re-repair failed (1) with logsectlog = random.
re-repair failed (1) with logsectsize = zeroes.
re-repair failed (1) with logsectsize = ones.
re-repair failed (1) with logsectsize = firstbit.
re-repair failed (1) with logsectsize = middlebit.
re-repair failed (1) with logsectsize = lastbit.
re-repair failed (1) with logsectsize = add.
re-repair failed (1) with logsectsize = random.
re-repair failed (1) with logsunit = ones.
re-repair failed (1) with logsunit = firstbit.
re-repair failed (1) with logsunit = random.
re-repair failed (1) with features_incompat = zeroes.
re-repair failed (1) with features_incompat = add.
re-repair failed (1) with crc = zeroes.
re-repair failed (1) with crc = ones.
re-repair failed (1) with crc = firstbit.
re-repair failed (1) with crc = middlebit.
re-repair failed (1) with crc = lastbit.
re-repair failed (1) with crc = add.
re-repair failed (1) with crc = sub.
re-repair failed (1) with crc = random.
re-repair failed (1) with spino_align = zeroes.
re-repair failed (1) with spino_align = ones.
re-repair failed (1) with spino_align = firstbit.
re-repair failed (1) with spino_align = middlebit.
re-repair failed (1) with spino_align = lastbit.
re-repair failed (1) with spino_align = add.
re-repair failed (1) with spino_align = sub.
re-repair failed (1) with spino_align = random.
re-repair failed (1) with pquotino = firstbit.
re-repair failed (1) with pquotino = middlebit.
re-repair failed (1) with pquotino = lastbit.
re-repair failed (1) with pquotino = add.
re-repair failed (1) with pquotino = sub.
re-repair failed (1) with pquotino = random.
re-repair failed (1) with meta_uuid = ones.
re-repair failed (1) with meta_uuid = firstbit.
re-repair failed (1) with meta_uuid = middlebit.
re-repair failed (1) with meta_uuid = lastbit.
re-repair failed (1) with meta_uuid = random.
re-repair failed (1) with rrmapino = ones.
re-repair failed (1) with rrmapino = firstbit.
re-repair failed (1) with rrmapino = middlebit.
re-repair failed (1) with rrmapino = lastbit.
re-repair failed (1) with rrmapino = add.
re-repair failed (1) with rrmapino = sub.
re-repair failed (1) with rrmapino = random.
Done fuzzing superblock
